ppx/melange_ppx.ml

```diff
@@ -954,6 +951,28 @@ module Derivers = struct
     in
     Deriving.add ~str_type_decl ~sig_type_decl "abstract"
```

Comment:
What do you think would be the plans for `abstract`? Would it be deprecated in another PR?

Reply:
yeah, I suppose we should probably deprecate it if we're going ahead with this. as I mentioned in https://github.com/melange-re/melange/pull/987#discussion_r1428095102, I'll implement abstract in terms of the new functions too.